### ISSUE TITLE
Custom handlers broken in v2.11.2

### ISSUE THREAD

**Issue by fridolin-koch** [symfony/monolog-bundle#181]
Hi,

after upgrading from v2.11.1 to v2.11.2, custom handlers stopped working.

```
[InvalidArgumentException]
  There is no handler class defined for handler "service".

Exception trace:
 () at /path/vendor/symfony/monolog-bundle/DependencyInjection/MonologExtension.php:727
 Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension->getHandlerClassByType() at /path/vendor/symfony/monolog-bundle/DependencyInjection/MonologExtension.php:133
 Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension->buildHandler() at /path/vendor/symfony/monolog-bundle/DependencyInjection/MonologExtension.php:68
 Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension->load() at /path/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php:55
 Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationPass->process() at /path/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/MergeExtensionConfigurationPass.php:39
 Symfony\Component\HttpKernel\DependencyInjection\MergeExtensionConfigurationPass->process() at /path/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php:104
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /path/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:528
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /path/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:477
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /path/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:117
 Symfony\Component\HttpKernel\Kernel->boot() at /path/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:68
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /path/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:118
 Symfony\Component\Console\Application->run() at /path/bin/console:27
```

Here is how the handler is defined:

```yaml
#monolog
monolog:
  channels: ["my_project"]
  handlers:
    notification:
      type: service
      id: my_project.monolog.handler.notification
      level: info
      channels: my_project
```
Downgrading to `2.11.1` solves the issue.

**Comment by avant1** [comment]
Thats really strange that `2.11.2` was released since it contains BC break (#170).

But anyway, the described issue is a bug that was introduced in aforementioned PR, i'll work on a fix for it.

**Comment by fabpot** [comment]
Sorry about that. Just released 2.11.3.

**Comment by fridolin-koch** [comment]
Thanks for the quick response :) 👍
